src/node_kernels/MomentumSSTTAMSForcingNodeKernel.C

```diff
@@ -0,0 +1,237 @@
+  NALU_ALIGNED NodeKernelTraits::DblType fluctU[NodeKernelTraits::NDimMax]; // fluctuatingVelocity
+
+  const NodeKernelTraits::DblType dualVolume = dualNodalVolume_.get(node, 0);
+
+  const NodeKernelTraits::DblType mu = viscosity_.get(node, 0);
+  const NodeKernelTraits::DblType tvisc = tvisc_.get(node, 0);
+  const NodeKernelTraits::DblType rho = density_.get(node, 0);
+  const NodeKernelTraits::DblType tke = stk::math::max(tke_.get(node, 0), 1.0e-12);
+  const NodeKernelTraits::DblType sdr = sdr_.get(node, 0);
+  const NodeKernelTraits::DblType alpha = alpha_.get(node, 0);
+  const NodeKernelTraits::DblType wallDist = minDist_.get(node, 0);
+  const NodeKernelTraits::DblType avgRho = avgDensity_.get(node, 0);
+  const NodeKernelTraits::DblType avgResAdeq = avgResAdeq_.get(node, 0);
+
+  for (int d = 0; d < nDim_; d++) {
+    avgU[d] = avgVelocity_.get(node, d);
+    fluctU[d] = velocity_.get(node, d) - avgVelocity_.get(node, d);
+    coords[d] = coordinates_.get(node, d);
+  }
+
+  const NodeKernelTraits::DblType eps = betaStar_ * tke * sdr;
+
+  // First we calculate the a_i's
+  const double FORCING_CL = 4.0;
+  const double Ceta = 70.0;
+  const double Ct = 6.0;
+  const double BL_T = 1.0;
+  const double BL_KOL = 1.0;
+  const double FORCING_FACTOR = forceFactor_;
+
+  const NodeKernelTraits::DblType periodicForcingLengthX = pi_;
+  const NodeKernelTraits::DblType periodicForcingLengthY = 0.25;
+  const NodeKernelTraits::DblType periodicForcingLengthZ = 3.0 / 8.0 * pi_;
+
+  NodeKernelTraits::DblType length =
+    FORCING_CL * stk::math::pow(alpha * tke, 1.5) / eps;
+  length = stk::math::max(
+    length, Ceta * (stk::math::pow(mu/rho, 0.75) / stk::math::pow(eps, 0.25)));
+  length = stk::math::min(length, wallDist);
+
+  NodeKernelTraits::DblType T_alpha = alpha * tke / eps;
+  T_alpha = stk::math::max(T_alpha, Ct * stk::math::sqrt(mu / rho / eps));
+  T_alpha = BL_T * T_alpha;
+
+  const NodeKernelTraits::DblType Mij_00 = Mij_.get(node, 0);
+  const NodeKernelTraits::DblType Mij_11 = Mij_.get(node, 4);
+  const NodeKernelTraits::DblType Mij_22 = Mij_.get(node, 8);
+  const NodeKernelTraits::DblType ceilLengthX =
+    stk::math::max(length, 2.0 * Mij_00);
+  const NodeKernelTraits::DblType ceilLengthY =
+    stk::math::max(length, 2.0 * Mij_11);
+  const NodeKernelTraits::DblType ceilLengthZ =
+    stk::math::max(length, 2.0 * Mij_22);
+
+  const NodeKernelTraits::DblType clipLengthX =
+    stk::math::min(ceilLengthX, periodicForcingLengthX);
+  const NodeKernelTraits::DblType clipLengthY =
+    stk::math::min(ceilLengthY, periodicForcingLengthY);
+  const NodeKernelTraits::DblType clipLengthZ =
+    stk::math::min(ceilLengthZ, periodicForcingLengthZ);
+
+  // FIXME: Hack to do a round/floor/ceil/mod operation since it isnt in
+  // stk::math right now
+  NodeKernelTraits::DblType ratioX;
+  NodeKernelTraits::DblType ratioY;
+  NodeKernelTraits::DblType ratioZ;
+  for (int simdIndex = 0; simdIndex < stk::simd::ndoubles; ++simdIndex) {
```

Comment:
NodeKernels are currently plain `double` we can remove these simd loops

Reply:
done